prometheus_client/parser.py

```diff
@@ -19,129 +19,99 @@ def text_string_to_metric_families(text):
     for metric_family in text_fd_to_metric_families(StringIO.StringIO(text)):
         yield metric_family
 
+def _replace_help_escaping(s):
+    return s.replace("\\n", "\n").replace('\\\\', '\\')
 
-def _unescape_help(text):
-    result = []
-    slash = False
+def _replace_escaping(s):
+    return s.replace("\\n", "\n").replace('\\\\', '\\').replace('\\"', '"')
 
-    for char in text:
-        if slash:
-            if char == '\\':
-                result.append('\\')
-            elif char == 'n':
-                result.append('\n')
-            else:
-                result.append('\\' + char)
-            slash = False
-        else:
-            if char == '\\':
-                slash = True
-            else:
-                result.append(char)
 
-    if slash:
-        result.append('\\')
-
-    return ''.join(result)
+def _parse_labels(labels_string):
+    labels = {}
+    # Return if we don't have valid labels
+    if "=" not in labels_string:
+        return labels
+
+    escaping = False
+    if "\\" in labels_string:
+        escaping = True
+
+    # Copy original labels
+    sub_labels = labels_string
+    try:
+        # Process one label at a time
+        while sub_labels:
+            # The label name is before the equal
+            value_start = sub_labels.index("=")
+            label_name = sub_labels[:value_start]
+            sub_labels = sub_labels[value_start + 1:].lstrip()
+            # Find the first quote after the equal
+            quote_start = sub_labels.index('"') + 1
+            value_substr = sub_labels[quote_start:]
+
+            # Find the last unescaped quote
+            i = 0
+            while i < len(value_substr):
+                i = value_substr.index('"', i)
+                if value_substr[i - 1] != "\\":
```

Comment:
What if if you have `x=""` as the label? `i - 1` will be -1, which might have unexpected results

Reply:
added a unit-test around empty labels, but this works fine 👍